Milyli.ScriptRunner.Core/Tools/RelativityScriptProcessor.cs

```diff
@@ -0,0 +1,163 @@
+﻿namespace Milyli.ScriptRunner.Core.Tools
+{
+	using System;
+	using System.Collections.Generic;
+	using System.Linq;
+	using System.Text.RegularExpressions;
+	using global::Relativity.API;
+	using kCura.Relativity.Client;
+	using Milyli.ScriptRunner.Core.Models;
+
+	/// <inheritdoc />
+	public class RelativityScriptProcessor : IRelativityScriptProcessor
+	{
+		private static List<string> textDataTypes = new List<string>
+		{
+			"date",
+			"datetime",
+			"text"
+		};
+
+		private readonly IHelper relativityHelper;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="RelativityScriptProcessor"/> class.
+		/// </summary>
+		/// <param name="relativityHelper"><see cref="IHelper"/> dependency from Relativity.</param>
+		public RelativityScriptProcessor(IHelper relativityHelper)
+		{
+			this.relativityHelper = relativityHelper ?? throw new ArgumentNullException(nameof(relativityHelper));
+		}
+
+		/// <summary>
+		/// Gets name for a generated table containing a list of documents in a saved search.
+		/// </summary>
+		/// <param name="tablePrepend">Prepend used when creating the table.</param>
+		/// <param name="searchId">Id of the saved search.</param>
+		/// <param name="scriptRunnerJobId">Id of the script runner job.</param>
+		/// <returns>Generated table for the specified saved search.</returns>
+		public static string GetSearchTableName(string tablePrepend, int searchId, int scriptRunnerJobId) => tablePrepend + "_" + searchId.ToString() + "_" + scriptRunnerJobId.ToString();

```

Comment:
Can we please add some entropy to this table name? A timestamp or guid would go a long way towards preventing nonsense when a script is run twice.